src/tree.rs

```diff
@@ -436,6 +438,63 @@ impl<T> Tree<T> {
     }
 
     ///
+    /// Swaps two `Node`s including their children given their `NodeId`s.
+    ///
+    /// Returns an empty `Result` containing a `NodeIdError` if one occurred on either provided `Id`.
+    ///
+    /// Notes:
+    ///     - Both `NodeId`s are still valid after this process and are not swapped.
+    ///     - Keeps the positions of the `Node`s in their parents children collection.
+    ///
+    /// ``
```

Comment:
swap(B, C):
```
         A
       /   \
      B     C
     / \   / \
    D   E F   G
```
```
         A
       /   \
      C     B
     / \   / \
    F   G D   E
```

swap(C, F):
```
         A
      /     \
    B         C
   / \       / \
  D   E     F   G
 / \ / \   / \ / \
H  I J  K L  M N  O
```
```
         A
      /     \
    B         F
   / \       / \
  D   E     L   M
 / \ / \
H  I J  K
```

This is what carefully documented I would like to see.
But I get that it is not exactly intuitive. Another option would be to add C into F's children minus the F sub tree, but that is more complicated to do.

The other way around (move upper `Node` into its own sub tree - swap(F, C) [order would become important here]) should error I my opinion. If we go with the alternative approach we could also lift the children of C and add C as child of F.